### ISSUE TITLE
Import not resolved when creating python virtual env with existing site-package

### ISSUE THREAD

**Issue by Bracewind** [microsoft/pyright#1853]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**

I am currently trying to integrate an existing system site package inside a virtual environment in vscode. When I create a virtual environment and add package to it using pip, the imports are correctly recognized and everything works fine. However, when I add an existing site package, the packages found seems to be the packages of the basic python of my system (/bin/usr/python) and not the ones of the virtualenv anymore. Moreover, when I install other packages in the virtualenv using pip they are not recognized anymore by pyright. Notice however that when I open my terminal, source the virtual env and execute the command line tool pyright, the imports are correctly resolved. 

As such, I suppose the issue is in my vscode settings file.

**To Reproduce**

create a virtual environment using 
```
python3 -m venv --system-site-package <path-to-your-site-package> tvenv
```
open a new folder containing your code in VS code
write a settings.json config file for vscode
```
{
    "python.autoComplete.extraPaths": [
        "/home/venvs/tvenv/python3.8/site-packages"
    ],
    "python.analysis.extraPaths": [
        "/home/venvs/tvenv/python3.8/site-packages"
    ],
    "python.pythonPath": "/home/venvs/tvenv/bin/python",
}
```

write a .py file importing a package installed in the virtualenv and not in your global python environment

**Expected behavior**

no import error when creating a virtualenv using an existing site-packages.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.

A python file example used to test the extension
```python
import black     #black must not be installed in the user environment. It can be installed in the site package 
                         # or from pip after the virtual env was created
```
In that snippet, pyright returns an import error

**VS Code extension or command-line**

Everything works when running pyright as a command line tool. When integrating it with vs code using the described settings.json and the vs code extension pyright (v1.1.138), an import error appear. Pyright returns the following output when parsing files : Exception received when installing recursive file system watcher. But it seems to also be raised when not using site-package, so i do not think this is the issue.

**Comment by erictraut** [comment]
I recommend deleting all three of the settings in your settings.json file.

* "python.autoComplete.extraPaths" is obsolete and isn't used at all by pylance or pyright.
* "python.pythonPath" is deprecated.
* "python.analysis.extraPaths" should not be needed in this case.

If you do not have the Microsoft "Python" extension installed in VS Code, install it. It provides a way to select the Python environment for your workspace. Simply select the venv you want to use, and pylance/pyright will honor that setting.

**Comment by Bracewind** [comment]
I tried removing the settings, and selecting the venv, it indeed does not change anything and everything works perfectly when not using the site-package configuration.

RIght now, I do not know if the issue is due to the use of --system-site-package parameter or if it is my specific site-package that creates issues. However, the python exension correctly resolve the import, which is what I find really strange. I do not have time right now to do more tests.

**Comment by Bracewind** [comment]
I solved my issue, I did not understand really well what site packages were and this is why I had this issue. --system-site-package retrieves the global python package, which was not the behaviour I wanted. If you want to make use of a given site package folder, just create a requirement file using pip --freeze or copy the packages if they are not found on pypi in <name of your virtualenv>/lib/python3.8/site-packages folder.
